kratos/utilities/variable_redistribution_utility.cpp

```diff
@@ -12,115 +12,312 @@
 //
 
 #include "includes/fsi_variables.h"
+#include "utilities/atomic_utilities.h"
+#include "utilities/parallel_utilities.h"
 #include "variable_redistribution_utility.h"
 
 namespace Kratos
 {
 
+// Historical API
+// ConvertDistributedValuesToPoint for double variables
+KRATOS_DEPRECATED_MESSAGE("This ConvertDistributedValuesToPoint() signature is deprecated. Use the one with Elements/Conditions container.")
 void VariableRedistributionUtility::ConvertDistributedValuesToPoint(
     ModelPart& rModelPart,
     const Variable<double>& rDistributedVariable,
     const Variable<double>& rPointVariable)
 {
-    CallSpecializedConvertDistributedValuesToPoint(rModelPart,rDistributedVariable,rPointVariable);
+    CallSpecializedConvertDistributedValuesToPoint<true>(rModelPart,rModelPart.Conditions(),rDistributedVariable,rPointVariable);
 }
 
+void VariableRedistributionUtility::ConvertDistributedValuesToPoint(
+    ModelPart& rModelPart,
+    ModelPart::ConditionsContainerType& rConditions,
+    const Variable<double>& rDistributedVariable,
+    const Variable<double>& rPointVariable)
+{
+    CallSpecializedConvertDistributedValuesToPoint<true>(rModelPart,rConditions,rDistributedVariable,rPointVariable);
+}
+
+void VariableRedistributionUtility::ConvertDistributedValuesToPoint(
+    ModelPart& rModelPart,
+    ModelPart::ElementsContainerType& rElements,
+    const Variable<double>& rDistributedVariable,
+    const Variable<double>& rPointVariable)
+{
+    CallSpecializedConvertDistributedValuesToPoint<true>(rModelPart,rElements,rDistributedVariable,rPointVariable);
+}
+
+// ConvertDistributedValuesToPoint for array variables
+KRATOS_DEPRECATED_MESSAGE("This ConvertDistributedValuesToPoint() signature is deprecated. Use the one with Elements/Conditions container.")
```

Comment:
are you exposing this function? If so then consider changing this deprecation warning to runtime only, this should be more effective in this case since afaik the usages are through python

Reply:
Yes, the function is exposed. Changing it!

Reply:
Done.